test/src/test/java/jenkins/security/StackTraceSuppressionTest.java

```diff
@@ -0,0 +1,202 @@
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+package jenkins.security;
+
+import com.gargoylesoftware.htmlunit.html.HtmlPage;
+import hudson.model.FreeStyleProject;
+import hudson.model.ItemGroup;
+import hudson.model.TopLevelItemDescriptor;
+import hudson.model.User;
+import jenkins.model.Jenkins;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.jvnet.hudson.test.JenkinsRule;
+import org.jvnet.hudson.test.MockAuthorizationStrategy;
+
+import java.io.IOException;
+
+import static org.hamcrest.CoreMatchers.containsString;
+import static org.hamcrest.CoreMatchers.not;
+import static org.hamcrest.MatcherAssert.assertThat;
+
+public class StackTraceSuppressionTest {
+
+    @Rule
+    public JenkinsRule j = new JenkinsRule();
+
+    @Before
+    public void setup() {
+        Jenkins.SHOW_STACK_TRACE = false;
+    }
+
+    @Test
+    public void authenticationManageException() throws Exception {
+        j.jenkins.setSecurityRealm(j.createDummySecurityRealm());
+        j.jenkins.setAuthorizationStrategy(new MockAuthorizationStrategy().grant(Jenkins.READ).everywhere().to("alice"));
+        User alice = User.getById("alice", true);
+        JenkinsRule.WebClient wc = j.createWebClient();
+        wc.login(alice.getId());
+
+        wc.setThrowExceptionOnFailingStatusCode(false);
+        HtmlPage page = wc.goTo("manage");
+
+        String content = page.getWebResponse().getContentAsString();
+        assertThat(content, containsString(alice.getId() + " is missing the Overall/Administer permission"));
+        assertThat(content, not(containsString("Caused by")));
+    }
+
+    @Test
+    public void authenticationConfigureSecurityException() throws Exception {
+        j.jenkins.setSecurityRealm(j.createDummySecurityRealm());
+        j.jenkins.setAuthorizationStrategy(new MockAuthorizationStrategy().grant(Jenkins.READ).everywhere().to("alice"));
+        User alice = User.getById("alice", true);
+        JenkinsRule.WebClient wc = j.createWebClient();
+        wc.login(alice.getId());
+
+        wc.setThrowExceptionOnFailingStatusCode(false);
+        HtmlPage page = wc.goTo("configureSecurity");
+
+        String content = page.getWebResponse().getContentAsString();
+        assertThat(content, containsString(alice.getId() + " is missing the Overall/Administer permission"));
+        assertThat(content, not(containsString("Caused by")));
+    }
+
+//    @Test
```

Comment:
do you plan to remove these commented out tests?

Reply:
I'm thinking of enabling them again once incrementals publishing is fixed and I can get a Stapler incremental. They really belong in Stapler but they're a change and they're easier to test here.

Reply:
incremental publishing was fixed yesterday

Reply:
It wasn't when I tried yesterday but it is now. I've updated to use the Stapler incremental.